controllers/clusters/kafka_controller.go

```diff
@@ -234,9 +236,183 @@ func (r *KafkaReconciler) handleUpdateCluster(
 		return models.ReconcileRequeue
 	}
 
+	if len(k.Spec.UserRefs) > 0 {
+		for _, ref := range k.Spec.UserRefs {
+			err = r.handleCreateUser(ctx, k, l, ref)
+			if err != nil {
+				l.Error(err, "Cannot create Kafka user", "user", ref)
+				r.EventRecorder.Eventf(k, models.Warning, models.CreatingEvent,
+					"Cannot create user. Reason: %v", err)
+			}
+		}
+	}
```

Comment:
You can omit here checking the length of `k.Spec.UserRefs` because `for range` does it always.
```suggestion
		for _, ref := range k.Spec.UserRefs {
			err = r.handleCreateUser(ctx, k, l, ref)
			if err != nil {
				l.Error(err, "Cannot create Kafka user", "user", ref)
				r.EventRecorder.Eventf(k, models.Warning, models.CreatingEvent,
					"Cannot create user. Reason: %v", err)
			}
		}
```

Reply:
done